examples/decoder/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+#![no_std]
+#![feature(default_alloc_error_handler)]
+
+use gstd::{msg, prelude::*};
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let new_msg = String::from_utf8(msg::load()).expect("Invalid message: should be utf-8");
+
+    let code: Vec<usize> = new_msg
+        .split(";")
+        .map(|v| {
+            v.parse::<usize>()
+                .expect("Not a number was sent in sequence")
+                - 1
+        })
+        .collect();
+
+    let nodes = &code.len() + 2;
+
+    let mut degrees = vec![1; nodes];
+    for vertex in &code {
+        degrees[*vertex] += 1;
+    }
+
+    let mut leaves = vec![];
+    for vertex in 0..nodes {
+        if degrees[vertex] == 1 {
+            leaves.push(vertex);
+        }
+    }
+
+    let handle = msg::send_init(msg::source(), b"Edges:", 0, 0);
+
+    for vertex in &code {
+        leaves.sort_unstable();
```

Comment:
don't use `sort_unstable` in wasm, it can bring some non-determinism at compile-time